### ISSUE TITLE
[REVIEW]: splot: visual analytics for spatial statistics

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1882]
**Submitting author:** @slumnitz (<a href="http://orcid.org/0000-0002-7007-5812">Stefanie Lumnitz</a>)
**Repository:** <a href="https://github.com/pysal/splot" target ="_blank">https://github.com/pysal/splot</a>
**Version:** v1.1.1
**Editor:** @leouieda
**Reviewer:** @ResidentMario, @martinfleis
**Archive:** <a href="https://doi.org/10.5281/zenodo.3724199" target="_blank">10.5281/zenodo.3724199</a>

## Status

[![status](https://joss.theoj.org/papers/59166a5cec24d54bafec741aa08c3ba8/status.svg)](https://joss.theoj.org/papers/59166a5cec24d54bafec741aa08c3ba8)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/59166a5cec24d54bafec741aa08c3ba8"><img src="https://joss.theoj.org/papers/59166a5cec24d54bafec741aa08c3ba8/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/59166a5cec24d54bafec741aa08c3ba8/status.svg)](https://joss.theoj.org/papers/59166a5cec24d54bafec741aa08c3ba8)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/pysal/splot" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@ResidentMario & @martinfleis, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @leouieda know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pysal/splot">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@slumnitz) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pysal/splot">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@slumnitz) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @ResidentMario, @martinfleis it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01882/joss.01882/10.21105.joss.01882.pdf)

**Comment by leouieda** [comment]
👋 @slumnits @ResidentMario, @martinfleis this is the review thread for the paper. All of our communications will happen here from now on.

Both reviewers have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `openjournals/joss-reviews#1882` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within about 4 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me here (@leouieda) or email me privately if you have any questions/concerns.

**Comment by martinfleis** [comment]
Hi,

I would like to close up my review. Majority of `splot`'s feature is currently focused on visualising the data from various PySAL sub-modules, which is extremely helpful in understanding their outputs. However, `splot` goes further than that (and has plans to do even more). A prime example is Value-by-alpha plotting feature. The default visual style of `splot` is very polished, and I would also say that there is no need to touch the defaults and plots can go directly to publications. That is not very common for scientific software.

During my review, I have opened a couple of issues (all linked above). Most of them are just recommendations, rather than critical issues. Before I tick Installation box, I would like to see pysal/splot#88 fixed as there are currently two versions of instructions, with those in the documentation failing. The second issue I would like to see resolved to tick the final box is pysal/splot#92, but that seems to be just a syntax issue.

To sum it up, `splot` is a valuable part of Python geospatial world and I will be watching closely its future development. My last recommendation is to have a detailed discussion with `geoplot` (already started in pysal/splot#21) to agree on the scope of both projects and avoid duplication of effort in the future.

**Comment by ResidentMario** [comment]
I'm done with my review. I don't have any hard blocking issues to add to those opened by @martinfleis.

But I will say that I have some qualms with the uneven documentation in `splot`. For example, one of the three tutorial notebooks, titled [Exploratory Analysis of Spatial Data](https://nbviewer.jupyter.org/github/pysal/splot/blob/master/notebooks/esda_morans_viz.ipynb), mentions the following statistical concepts without giving much explanation or linking the user to a reference on what these things are: Moran's I, local Moran, Donatns [sic?] variable, bivariate Moran statistic. Furthermore, [the tutorials](https://splot.readthedocs.io/en/latest/index.html#) only cover a small subset of the library functions: there are many plotters documented at some length in the API Reference which do not appear in the any of the tutorials.

I do not think this is a blocking issue because the documentation is still within the acceptable range for JOSS (all of the relevant methods are covered by the API Reference, which doesn't leave any fields unexplained and is generally very well-packed with example images), but I feel that it's a definite area for improvement for the package as a whole.

I have left more general feedback about documentation with the broader PySAL team in the [PySAL Gitter chat](https://gitter.im/pysal/pysal?at=5dd1dfd7e75b2d5a19efb7cc).

**Comment by leouieda** [comment]
Hi @martinfleis @ResidentMario thank you for your reviews! :tada: 

@ResidentMario that is a good point about the tutorial coverage and I agree that it's not a requirement for JOSS to have full coverage of tutorials. However, the citation/linking of statistical concepts is highly encouraged as I see scientific software documentation (particularly for JOSS) as a major part of the scholarship. @slumnitz could you please go through the tutorials and add references as you see necessary? 

@martinfleis @slumnitz please let me know when the install and figure caption issues are resolved.

**Comment by slumnitz** [comment]
@martinfleis @ResidentMario @leouieda thank you for your thorough review, comments and great suggestions! I will implement them in the next couple of weeks or as soon as I can and let you know when everything is done.

**Comment by danielskatz** [comment]
@slumnitz - any news on this?

**Comment by slumnitz** [comment]
> @slumnitz - any news on this?

@danielskatz I will start working on it slowly again now. It got a bit delayed due to snowmageddon and a couple of electricity outages here in BC, Canada, which are expected to ease in the next wo weeks.

**Comment by leouieda** [comment]
@slumnitz thank you for the update :+1: Please keep us informed of your progress.

**Comment by slumnitz** [comment]
@martinfleis @ResidentMario and @leouieda thank you for your review and comments. The new splot release includes solutions to the major documentation issues outlined. Currently as you noticed, splot tutorials require previous knowledge in spatial statistical analysis and are not aimed to teach statistical concepts. 

Do you think a new tutorial like `a beginners guide to spatial statistical analysis in splot` would be suitable to fill this gap, or would you recommend referencing great blogposts, statistical literature is enough?

**Comment by martinfleis** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01882/joss.01882/10.21105.joss.01882.pdf)

**Comment by martinfleis** [comment]
@slumnitz the figure captions formatting is still strange. JOSS has its own style for captions, see the [last published paper](https://joss.theoj.org/papers/10.21105/joss.01800). Your figures seems to be encapsulated in links, which are not working anyway, causing captions not to render.

**Comment by martinfleis** [comment]
@slumnitz to answer your question, I think good links are enough.

**Comment by leouieda** [comment]
@slumnitz thank you keeping us in the loop. As @martinfleis said, the figures weren't rendering properly. I submitted a PR fixing this. 

Also, we now have a compilation service that you can use to preview the paper (even on branches before merging PRs): https://whedon.theoj.org

@ResidentMario I see you still have the "Installation" review item unchecked. Could you please see if this has been addressed in the new version of splot?

@martinfleis you have the "Quality of writing" item unchecked. Has this been addressed in the updated paper?

**Comment by martinfleis** [comment]
@leouieda I kept it unchecked due to figures formatting issue. Your PR resolves it, but note that you made a PR to your forked repo not the original repo. 

I'll take it as resolved and finish my review.

**Comment by ResidentMario** [comment]
I had the installation mark unchecked due to splot#88, but that's now been fixed so I went ahead and checked the box.

[The new docs](http://pysal.org/notebooks/viz/splot/intro.html) look great!

That's everything from me.

**Comment by leouieda** [comment]
> Your PR resolves it, but note that you made a PR to your forked repo not the original repo.

Well, that was embarrassing :) Resolved now with pysal/splot#99

@martinfleis and @ResidentMario thank you for your reviews! 

@slumnitz I'll go over the submission one more time to make any editorial suggestions that might be needed. After that, I'm happy to move forward with acceptance of your submission to JOSS :tada:

**Comment by leouieda** [comment]
@whedon check references

**Comment by ljwolf** [comment]
The rerender of the paper after mering pysal/splot#99 is [here](https://res.cloudinary.com/hju22ue2k/image/upload/v1583329488/puu3xi4ibwtllubo20fg.pdf) and should conform to the requests, so let me or @slumnitz know if anything else is required.

**Comment by arfon** [comment]
Dear authors and reviewers

We wanted to notify you that in light of the current COVID-19 pandemic, JOSS has decided to suspend submission of new manuscripts and to handle existing manuscripts (such as this one) on a "best efforts basis". We understand that you may need to attend to more pressing issues than completing a review or updating a repository in response to a review. If this is the case, a quick note indicating that you need to put a "pause" on your involvement with a review would be appreciated but is not required.

Thanks in advance for your understanding.

_Arfon Smith, Editor in Chief, on behalf of the JOSS editorial team._

**Comment by ljwolf** [comment]
Thanks @arfon, I understand this is a challenging time for everyone. I believe the submission is complete; @leouieda's suggestions were the last ones remaining, and have been addressed.

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by arfon** [comment]
@ljwolf - I'm not sure of @leouieda's availability to handle the final steps here so...

At this point could you make a new release of this software that includes the changes that have resulted from this review. Then, please make an archive of the software in Zenodo/figshare/other service and update this thread with the DOI of the archive? For the Zenodo/figshare archive, please make sure that:

- The title of the archive is the same as the JOSS paper title
- That the authors of the archive are the same as the JOSS paper authors

I can then move forward with accepting the submission.

**Comment by ljwolf** [comment]
Will do. Thanks!

**Comment by slumnitz** [comment]
@ljwolf thanks for taking this on. I finally have a stable wifi connection again here in Italy. I will take this on tonight. Would you be able to review as soon as I made the PR's?

**Comment by slumnitz** [comment]
@arfon thank you for helping. Splot has a new release v.1.1.3 on Zenodo

DOI: 10.5281/zenodo.3724199

I am not sure how I can update this thread. Maybe you could help? Thank you.

**Comment by slumnitz** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.3724199 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3724199" target="_blank">10.5281/zenodo.3724199</a> is the archive.

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01882/joss.01882/10.21105.joss.01882.pdf)

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1882 with the following error: 

 E, [2020-03-23 14:07:26#418] ERROR -- : Failed to parse BibTeX on value "--140" (NAME) [#<BibTeX::Bibliography data=[12]>, "@", #<BibTeX::Entry >, {:author=>["Robert E Roth and Andrew W Woodruff and Zachary F Johnson"], :title=>["Value-by-alpha maps: {An} alternative to the cartogram"], :journal=>["The Cartographic Journal"], :year=>["2010"], :volume=>"47", :issue=>"2", :pages=>"130"}]
bibtex.y:138:in `on_error': Failed to parse BibTeX on value "--140" (NAME) [#<BibTeX::Bibliography data=[12]>, "@", #<BibTeX::Entry >, {:author=>["Robert E Roth and Andrew W Woodruff and Zachary F Johnson"], :title=>["Value-by-alpha maps: {An} alternative to the cartogram"], :journal=>["The Cartographic Journal"], :year=>["2010"], :volume=>"47", :issue=>"2", :pages=>"130"}] (BibTeX::ParseError)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/racc/parser.rb:259:in `_racc_do_parse_c'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/racc/parser.rb:259:in `do_parse'
	from bibtex.y:111:in `parse'
	from /app/vendor/bundle/ruby/2.4.0/gems/bibtex-ruby-5.1.2/lib/bibtex/bibliography.rb:67:in `parse'
	from /app/vendor/bundle/ruby/2.4.0/gems/bibtex-ruby-5.1.2/lib/bibtex/bibliography.rb:50:in `open'
	from /app/vendor/bundle/ruby/2.4.0/gems/bibtex-ruby-5.1.2/lib/bibtex/utilities.rb:25:in `open'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/lib/whedon/bibtex_parser.rb:36:in `generate_citations'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/lib/whedon/compilers.rb:245:in `crossref_from_markdown'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/lib/whedon/compilers.rb:21:in `generate_crossref'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/lib/whedon/processor.rb:95:in `compile'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/bin/whedon:79:in `compile'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/bin/whedon:116:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by arfon** [comment]
@slumnitz - this PR should fix the BibTeX issue here: https://github.com/pysal/splot/pull/104

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1007/978-3-319-50590-9_17 is OK
- 10.1016/j.apgeog.2013.09.012 is OK
- 10.1007/s10109-009-0086-8 is OK
- 10.1007/978-3-319-59511-5_2 is OK
- 10.5281/zenodo.3333010 is OK
- 10.5281/zenodo.3475569 is OK

MISSING DOIs

- https://doi.org/10.1080/01944360802146329 may be missing for title: The Ghost Map: The Story of London’s Most Terrifying Epidemic–and How It Changed Science, Cities, and the Modern World
- https://doi.org/10.1111/j.1435-5597.1988.tb01155.x may be missing for title: Do Spatial Effects Really Matter in Regression Analysis?

INVALID DOIs

- 10.1179/000870409X124887534553372 is INVALID
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1388

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1388, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@slumnitz - could you take a look at the [DOI summary above](https://github.com/openjournals/joss-reviews/issues/1882#issuecomment-602641085) to check to see if the DOIs under `MISSING` are correct recommendations and also check the `INVALID` DOI?

**Comment by slumnitz** [comment]
@arfon all missing doi's are added and the incorrect one was fixed.

**Comment by arfon** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1080/01944360802146329 is OK
- 10.1007/978-3-319-50590-9_17 is OK
- 10.1016/j.apgeog.2013.09.012 is OK
- 10.1111/j.1435-5597.1988.tb01155.x is OK
- 10.1007/s10109-009-0086-8 is OK
- 10.1007/978-3-319-59511-5_2 is OK
- 10.5281/zenodo.3333010 is OK
- 10.5281/zenodo.3475569 is OK
- 10.1179/000870409X12488753453372 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1242110671027941377) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1389
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01882](https://doi.org/10.21105/joss.01882)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@ResidentMario, @martinfleis - many thanks for your reviews here and to @leouieda for editing this submission ✨ 

@slumnitz - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01882/status.svg)](https://doi.org/10.21105/joss.01882)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01882">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01882/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01882/status.svg
   :target: https://doi.org/10.21105/joss.01882
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01882/status.svg)](https://doi.org/10.21105/joss.01882)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by slumnitz** [comment]
Thank you for your help and support @leouieda, @martinfleis, @ResidentMario and @arfon.

**Comment by leouieda** [comment]
@slumnitz congratulations on the publication! Huge thanks to @martinfleis and @ResidentMario for the reviews and @arfon for picking up where I left off (apologies for the disappearance).
